docs/examples/example_simple_oosm.py

```diff
@@ -0,0 +1,283 @@
+
+# Create a list of timesteps
+timestamps = [start_time]
+
+# create a set of truths
+truths = set()
+
+# Instantiate the groundtruth for the first target
+truth = GroundTruthPath([GroundTruthState([0, 1, -100, 0.3], timestamp=start_time)])
+
+for k in range(1, num_steps):
+    truth.append(GroundTruthState(
+        transition_model.function(truth[k-1], noise=True,
+                                  time_interval=timedelta(seconds=2)),
+        timestamp=start_time + timedelta(seconds=5*k)))
+    timestamps.append(truth.timestamp)
+
+truths.add(truth)
+
+# Create the measurements models
+from stonesoup.models.measurement.linear import LinearGaussian
+
+measurement_model = LinearGaussian(
+    ndim_state=4,
+    mapping=(0, 2),
+    noise_covar=np.diag([50, 50]))
+
+# Collect the measurements using scans
+scans = []
+
+# Create the detections
+from stonesoup.types.detection import TrueDetection
+
+# Loop over the timesteps and collect the detections
+for k in range(num_steps):
+    detections = set()
+
+    # Introduce the delay
+    if k%5==0:
+        delay = 25
+    else:
+        delay = 0
+
+    measurement = measurement_model.function(truth[k], noise=True)
+    detections.add(TrueDetection(state_vector=measurement,
+                                  groundtruth_path=truth,
+                                  timestamp=truth[k].timestamp))
+
+    # Scans for tracking and reordering
+    scans.append((truth[k].timestamp + timedelta(seconds=delay), detections))
+
+# Reorder the scans by their arrival time
+arrival_time_ordered = sorted(scans, key=lambda dscan: dscan[0])
+
+# %%
+# 2. Instantiate the tracking components;
+# ---------------------------------------
+# We have the scans containing the detections ordered by their arrival time.
+# It is time to prepare the tracking components.
+# In this simple example we employ a :class:`~.KalmanPredictor`
+# and :class:`~.KalmanUpdater` to perform the tracking.
+#
+
+from stonesoup.updater.kalman import KalmanUpdater
+from stonesoup.predictor.kalman import KalmanPredictor
+predictor = KalmanPredictor(transition_model)
+updater = KalmanUpdater(measurement_model)
+
+# create the prior
+from stonesoup.types.state import GaussianState
+
+prior = GaussianState(state_vector=[0, 1, -100, 0.3],
+                      covar=np.diag([1, 1, 1, 1]),
+                      timestamp=start_time)
+
+# duplicate the prior
+prior_lag = deepcopy(prior)
+
+# %%
+# 3. Run the tracker and visualise the results.
+# ---------------------------------------------
+# We have the detections and the tracking components ready to be used.
+#
+# It is known that some detections arrived with a fixed delay, therefore we create a
+# buffer storage of fixed dimension (:math:`\ell`) where we store the detections and where we check the
+# timestamps and adjust them to their correct order. When one detection comes with an
+# earlier timestamp (:math:`\tau < t_{k}`), we shuffle the detections to keep the chain of detections.
```

Comment:
I think it would read better if this paragraph moves under the header "Let's deal with OOS detections", to keep the explanation near the relevant part of code.